server/blob/blob-export-file/src/test/java/org/apache/james/blob/export/file/LocalFileBlobExportMechanismTest.java

```diff
@@ -132,4 +137,32 @@ void shareShouldFailWhenBlobDoesNotExist() {
                 .export())
             .isInstanceOf(ObjectStoreException.class);
     }
+
+    @Nested
+    class ConfigurationTest {
```

Comment:
what if the string is the path of a file and instead of a directory ?
etc ...

Reply:
to check that, I need to create a file instance, then I will come back to the issue of having fileSystem instance to able to get a file from input string. Otherwise, from a string `file://var/blob`, I don't know whether it's a file or directory.

Reply:
IMO We should move these two lines:

```
            File exportingDirectory = fileSystem.getFile(configuration.exportDirectory);
```

From LocalFileBlobExportMechanism::copyBlobToFile to the constructor & add a check that it is not a file in the constructor.

Note: we should have a similar check in `copyBlobToFile` + the forceMkdir as well to avoid errors if the directory is deleted for instance.

So that we get an exception at app startup and not everytime you share something...